src/OnMessageTask.tcc

```diff
@@ -9,7 +9,7 @@
 
 template <typename T>
 class GeneralMessageTask : public OnMessageTask {
-    tbb::task* execute() {
+    OnMessageTask* execute() {
```

Comment:
Can this just be a void or bool? Most of the time we just returned a nullptr in our code

Reply:
Possibly it can but I want this code tested and wanted to do as little changes before that was done. We can't afford to have this update go missing again. It is more than 1.5 years since I first wrote this code!